File: Server/aoRss/Models/Domain/_blankModel.vb
﻿
Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models.Domain
    Public Class xxxxxmodelNameGoesHerexxxxx
        Inherits baseDomainModel
        '
        '====================================================================================================
        '-- const
        Private Shadows Const contentDataSource As String = "default"             '<------ set to datasource if not default
        '
        '====================================================================================================
        ' -- instance properties
        Public Property personName As String
        Public Property organizationName As String
        '
        '====================================================================================================
        ''' <summary>
        ''' get a list of objects matching the organizationId
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="organizationId"></param>
        ''' <param name="pageSize"></param>
        ''' <param name="pageNumber"></param>
        ''' <returns></returns>
        Public Overloads Shared Function createList(cp As CPBaseClass, organizationId As Integer, Optional pageSize As Integer = 999999, Optional pageNumber As Integer = 1) As List(Of xxxxxmodelNameGoesHerexxxxx)
            Dim sql As String = My.Resources.sampleSql.Replace("{0}", organizationId.ToString())
            Return createListFromSql(Of xxxxxmodelNameGoesHerexxxxx)(cp, sql, pageSize, pageNumber)
        End Function

    End Class
End Namespace

File: Server/aoRss/Models/Domain/_blankModel.cs




using System.Collections.Generic;
using Contensive.BaseClasses;

namespace Contensive.Addons.Rss.Models.Domain {
    public class xxxxxmodelNameGoesHerexxxxx : baseDomainModel {
        // 
        // ====================================================================================================
        // -- const
        private new const string contentDataSource = "default";             // <------ set to datasource if not default
        // 
        // ====================================================================================================
        // -- instance properties
        public string personName { get; set; }
        public string organizationName { get; set; }
        // 
        // ====================================================================================================
        /// <summary>
        /// get a list of objects matching the organizationId
        /// </summary>
        /// <param name="cp"></param>
        /// <param name="organizationId"></param>
        /// <param name="pageSize"></param>
        /// <param name="pageNumber"></param>
        /// <returns></returns>
        public static List<xxxxxmodelNameGoesHerexxxxx> createList(CPBaseClass cp, int organizationId, int pageSize = 999999, int pageNumber = 1) {
            string sql = My.Resources.Resources.sampleSql.Replace("{0}", organizationId.ToString());
            return createListFromSql<xxxxxmodelNameGoesHerexxxxx>(cp, sql, pageSize, pageNumber);
        }

    }
}